tests/integration/targets/module_utils_botocore_recorder/main.yml

```
@@ -0,0 +1,14 @@
+---
+- hosts: localhost
+  tasks:
+    - name: Call aws_az_info
+      amazon.aws.aws_az_info:
+      register: result
+    - debug: var=result
+    - name: Get called information
+      amazon.aws.aws_caller_info:
+      register: result
+    - debug: var=result
+    - assert:
+        that:
+         - lookup('ansible.builtin.env', '_ANSIBLE_PLACEBO_RECORD') or (lookup('ansible.builtin.env', '_ANSIBLE_PLACEBO_REPLAY') and result.user_id == "AWZBREIZHEOMABRONIFVGFS6GH")
```

```suggestion
    - assert:
        that:
        - '"user_id" in result'
        when:
        - lookup('ansible.builtin.env', '_ANSIBLE_PLACEBO_RECORD')
    - assert:
        that:
        - '"user_id" in result'
        # Once recorded we fully anonymize the UserID in the recording
        -  result.user_id == "AWZBREIZHEOMABRONIFVGFS6GH"
        when:        
        - lookup('ansible.builtin.env', '_ANSIBLE_PLACEBO_REPLAY')
    - assert:
        that:
        - False
        when:
        - not lookup('ansible.builtin.env', '_ANSIBLE_PLACEBO_REPLAY')
        - not lookup('ansible.builtin.env', '_ANSIBLE_PLACEBO_RECORD')
```

Do you think this is easier to follow as an example or am I over-thinking this?

---

I actually get this error `"Invalid options for assert: when"`. I will keep the original version.